glib/src/subclass/object.rs

```diff
@@ -224,54 +224,39 @@ mod test {
     use super::super::super::value::{ToValue, Value};
     use super::*;
     use crate::{StaticType, Type};
+    use crate as glib;
 
     use std::cell::RefCell;
 
     mod imp {
         use super::*;
 
         // A dummy `Object` to test setting an `Object` property and returning an `Object` in signals
+        #[derive(Default)]
```

Comment:
Does it only require `Default` if no `fn new()` or `fn with_class()` is provided in the `impl` block? That would be great

Reply:
Yes, that's the idea. The proc macro iterates through its input and sees if `new` or `with_class` are provided. If it finds neither, it adds a `new` definition.

The only issue is that I don't know if there's a way to print a nice warning specifically saying to implement `Default` or `new` (it errors saying the type doesn't implement `Default`, in the case where no `new` or `with_class` is defined, and `Default` isn't implemented).

Reply:
As macros don't know any type information that won't be possible. You can't know whether `Default` is implemented or not